distage/distage-testkit-core/src/main/scala/izumi/distage/testkit/services/dstest/DistageTestRunner.scala

```diff
@@ -73,10 +74,17 @@ class DistageTestRunner[F[_]: TagK]
         // here we find all the shared components in each of our individual tests
         val sharedKeys = testPlans.map(_._2).flatMap {
           plan =>
-            plan.steps.filter(env memoizationRoots _.target).map(_.target)
+            val memoized = plan.steps.filter(env memoizationRoots _.target).map(_.target)
+            val weak = plan.steps.collect {
+              // here we adding each weak reference into shared keys (we assume that if we have a weak reference in one test plan - it's should be bind)
+              case w: WiringOp.ReferenceKey if w.wiring.weak => w.target
```

Comment:
This makes ALL weak references into roots of the memoization plan - this is too broad and will result in a huge chunk of the graph being memoized.